### DISCUSSION TITLE
Multi Tenancy approach with a single client

### DISCUSSION THREAD

**Discussion by EireMan-dev** [fief-dev/fief#442] [Q&A]
Hi team. Firstly, thank you for providing such a great project!

As per your docs, a client can be attached to a single tenant only; which is appropriate and makes sense in a true multi-tenancy scenario. I hope to build a FastAPI/React service powered by Fief, for a number of customers whereby sub users for each customer are maintained as separate groups but managed together. I initially thought tenants would work, but since this will not be a white labelled product and controlled by my single fastAPI client, I can only have one tenant.

I am trying to see if there is a recommended way to manage multiple customer's users when done so through a centralised application. At the moment I am presuming a lookup table in my own db but I just want to confirm my understanding is correct, thanks!

**Comment by fief-bailiff** [comment]
Hail, @EireMan-dev 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
Hi @EireMan-dev 👋 

What you describe is what I may call "groups management". Indeed, Fief tenants won't be helpful for this use-case. I hope to add that feature in the future, but for now yes, you should come with your own logic for that.
